modules/platform/nuxeo-drive-server/nuxeo-drive-mongodb/src/main/java/org/nuxeo/drive/mongodb/MongoDBAuditChangeFinder.java

```diff
@@ -0,0 +1,232 @@
+                if (!allEntries.isEmpty()) {
+                    log.debug("Found no audit log entries matching the criterias but some exist, returning 0");
+                    return 0;
+                }
+            }
+            log.debug("Found no audit log entries, returning -1");
+            return -1;
+        }
+        return entries.get(0).getId();
+    }
+
+    @Override
+    protected List<LogEntry> queryAuditEntries(CoreSession session, SynchronizationRoots activeRoots,
+            Set<String> collectionSyncRootMemberIds, long lowerBound, long upperBound, int limit) {
+        MongoDBAuditBackend auditService = (MongoDBAuditBackend) Framework.getService(AuditReader.class);
+        MongoCollection<Document> auditCollection = auditService.getAuditCollection();
+        StringBuilder queryBuilder = new StringBuilder();
+        Map<String, Object> params = new HashMap<>();
+        params.put("repositoryId", session.getRepositoryName());
+        params.put("lowerBound", lowerBound);
+        params.put("upperBound", upperBound);
+
+        if(queryStart == null) {
+            try (InputStream is = getClass().getClassLoader().getResourceAsStream("OSGI-INF/queryStart.json")) {
+                queryStart = IOUtils.toString(is, "UTF-8");
+            } catch (IOException e) {
+                log.error("couldn't read query start template", e);
+            }
+        }
+        queryBuilder.append(queryStart);
+        // first conditional filter
+        if (!activeRoots.getPaths().isEmpty()) {
+            queryBuilder.append(",\n");
+            // We open an $or array here in case the next conditional filter is needed as it's this or that
+            queryBuilder.append("{ \"$or\" : [");
+            queryBuilder.append(getCurrentRootFilteringClause(activeRoots.getPaths(), params) + "\n");
+        }
+        // second conditional filter
+        if (collectionSyncRootMemberIds != null && !collectionSyncRootMemberIds.isEmpty()) {
+            queryBuilder.append(",\n");
+            queryBuilder.append(getCollectionSyncRootFilteringClause(collectionSyncRootMemberIds, params) + "\n");
+        }
+        // if the first conditional filter was set, then an $or array has been opened and needs to be closed
+        if (!activeRoots.getPaths().isEmpty()) {
+            queryBuilder.append("]}\n");
+        }
+        if(queryEnd == null) {
+            try (InputStream is = getClass().getClassLoader().getResourceAsStream("OSGI-INF/queryEnd.json")) {
+                queryEnd = IOUtils.toString(is, "UTF-8");
+            } catch (IOException e) {
+                log.error("couldn't read query end template", e);
+            }
+        }
+        queryBuilder.append(queryEnd);
+
+        Bson filter = auditService.buildFilter(queryBuilder.toString(), params);
+        var sorts = List.of(Sorts.ascending("repositoryId"), Sorts.descending("eventDate"));
```

Comment:
If this has to be backported to 10.10 you'd bette not use `var`, or be ready to update the backport for Java 8.